python/cudf/cudf/core/reshape.py

```diff
@@ -278,9 +338,24 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 result = objs[0].copy()
             return result
         else:
-            return cudf.DataFrame._concat(
-                objs, axis=axis, ignore_index=ignore_index, sort=sort
+            result = cudf.DataFrame._concat(
+                objs,
+                axis=axis,
+                join=join,
+                ignore_index=ignore_index,
+                sort=sort,
             )
+            if ignore_index and join == "inner" and result.empty:
+                result.index = cudf.RangeIndex(sum(len(obj) for obj in objs))
```

Comment:
I think this is tied into my comment above related to empty Index of `object` type vs RangeIndex